Fix mismatches between function parameter definitions and declarations

Troy-Butler opened the pull request:
Addresses issue #88716.

Some function parameter names in the affected header files did not match the parameter names in the definitions, or were listed in a different order.

Troy-Butler committed:
Fix Definition-Declaration Mismatches

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler committed:
Fix Code Formatting Issues

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler committed:
Remove trailing whitespace

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler commented:
I do not have the ability to merge, so if everything looks good with my pull request, can the last reviewer please merge this? Thank you!

NagyDonat commented on llvm/lib/Transforms/InstCombine/InstCombineInternal.h:
```diff
@@ -433,7 +433,7 @@ class LLVM_LIBRARY_VISIBILITY InstCombinerImpl final
                                        Instruction *CxtI, bool IsAnd,
                                        bool IsLogical = false);
-  Value *matchSelectFromAndOr(Value *A, Value *B, Value *C, Value *D,
+  Value *matchSelectFromAndOr(Value *A, Value *C, Value *B, Value *D,
```
What the <expletive> is happening here?

...I understand that this commit adjusts the declaration to match the names appearing in the definition, but why would the definition use the names "A", "C", "B" and "D" in this order?

This might be a bug in the definition of this function -- and even if it's not, I think it would be worth to preserve the natural ordering of these letters and adjust the definition of the function accordingly. (Of course, using descriptive variable names instead of meaningless single letters would be even better...)

Troy-Butler replied:
The pre-commit version of the function declaration listed the parameters in the order A, C, B, D to indicate the expected grouping of arguments, i.e. (A & C) | (B & D). Changing the order to A, B, C, D would have required reworking the function, which seemed outside the scope of this issue - so I opted to reflect the A, C, B, D order in the function declaration instead. However, as requested, I have modified the parameter names, listed them in logically ascending order, and reworked the function to match.

Troy-Butler committed:
Fix instCombineAndOrXor parameter names

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler committed:
Adjust code formatting

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler committed:
Adjust code formatting again

Signed-off-by: Troy-Butler <squintik@outlook.com>

NagyDonat commented:
> The pre-commit version of the function declaration listed the parameters in the order A, C, B, D to indicate the expected grouping of arguments, i.e. (A & C) | (B & D). Changing the order to A, B, C, D would have required reworking the function, which seemed outside the scope of this issue - so I opted to reflect the A, C, B, D order in the function declaration instead. However, as requested, I have modified the parameter names, listed them in logically ascending order, and reworked the function to match.

Thanks for explaining the situation and updating the code!

I think declaring the variable names in an alphabetical order will make the code less surprising and easier to read. However, now that I know more about the context where these variables are used, I think returning to the single-letter `A`, `B`, `C` and `D` would be better than the `valA`, `valB`, `valC`, `valD` that you introduced for two reasons:
- Previously I neglected to look at the definition of this function; now that I see that it's dealing with abstract Boolean predicates, I think that it's a good choice to use the single-letter names that are customary as mathematical notation.
- There is also a [design rule](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) which says that variables need to be capitalized, so the lowercase `val` prefix should not be introduced. (BTW this rule is not absolute: in existing code that uses a different naming convention, it's permissible/preferred to use that convention for the sake of consistency.)
I know that you probably introduced the `val` prefix because of my complaint about the "meaningless single letter" variable names; please excuse me for these back-and-forth suggestions.

Troy-Butler committed:
Revert instCombineAndOrXor parameter names

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler committed:
Code formatting adjustment

Signed-off-by: Troy-Butler <squintik@outlook.com>

Troy-Butler commented:
> > The pre-commit version of the function declaration listed the parameters in the order A, C, B, D to indicate the expected grouping of arguments, i.e. (A & C) | (B & D). Changing the order to A, B, C, D would have required reworking the function, which seemed outside the scope of this issue - so I opted to reflect the A, C, B, D order in the function declaration instead. However, as requested, I have modified the parameter names, listed them in logically ascending order, and reworked the function to match.
> 
> Thanks for explaining the situation and updating the code!
> 
> I think declaring the variable names in an alphabetical order will make the code less surprising and easier to read. However, now that I know more about the context where these variables are used, I think returning to the single-letter `A`, `B`, `C` and `D` would be better than the `valA`, `valB`, `valC`, `valD` that you introduced for two reasons:
> 
> * Previously I neglected to look at the definition of this function; now that I see that it's dealing with abstract Boolean predicates, I think that it's a good choice to use the single-letter names that are customary as mathematical notation.
> * There is also a [design rule](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) which says that variables need to be capitalized, so the lowercase `val` prefix should not be introduced. (BTW this rule is not absolute: in existing code that uses a different naming convention, it's permissible/preferred to use that convention for the sake of consistency.)
> 
> I know that you probably introduced the `val` prefix because of my complaint about the "meaningless single letter" variable names; please excuse me for these back-and-forth suggestions.

No worries, I understand. I've reverted the parameter variable names back to single letter form.

NagyDonat reviewed (approved):
Thanks for the update! I'll merge the PR within ~a day if nobody objects.

NagyDonat commented:
@Troy-Butler
I initiated the merge, but it seems that github mangles the email metadata:
![image](https://github.com/llvm/llvm-project/assets/43410265/d783c97f-f5f7-469f-9f77-9689275ea429)
(Note that I obscured your email address in this screenshot, but it can be found within the commits that you published.)

- If you want to use your "real" email address as the Author of this commit, then you probably need to change your github account settings. (I don't know the exact details but IIRC you can toggle whether it will use your email address in your commits.)
- If you don't want to embed that email address into the LLVM git history, then I can merge the commit without the `Signed-off-by` and `Co-authored-by` lines that mention it. The address will remain on the internet (in this pull request), but it won't be part of the LLVM main branch.
